hadoop-ozone/recon-codegen/src/main/java/org/hadoop/ozone/recon/codegen/ReconSqlDbConfig.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ * <p>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.hadoop.ozone.recon.codegen;
+
+import org.apache.hadoop.hdds.conf.Config;
+import org.apache.hadoop.hdds.conf.ConfigGroup;
+import org.apache.hadoop.hdds.conf.ConfigTag;
+import org.apache.hadoop.hdds.conf.ConfigType;
+
+/**
+ * The configuration class for the Recon SQL DB.
+ */
+@ConfigGroup(prefix = "ozone.recon.sql.db")
+public class ReconSqlDbConfig {
```

Comment:
Why does this class only provide 1 configuration item? The JooqPersistence module properties should not be moved here?

Reply:
Yes, but the Java configuration based API is for new config items generally. The code looks at ozone-default.xml and also Java based configuration classes and creates a master list for the default xml during assembly.

Reply:
I am 100% happy with the answer...

... but as @anuengineer is not here I feel the responsibility to decorate pull requests with random external literature. :smile: 

 Slow migration to Java based configuration API can happen at any time:

> THE BOY SCOUTS HAVE A RULE: “Always leave the campground cleaner than you found it.” If you find a mess on the ground, you clean it up regardless of who might have made it. You intentionally improve the environment for the next group of campers. (Actually, the original form of that rule, written by Robert Stephenson Smyth Baden-Powell, the father of scouting, was “Try and leave this world a little better than you found it.”)

> What if we followed a similar rule in our code: “Always check a module in cleaner than when you checked it out”? Regardless of who the original author was, what if we always made some effort, no matter how small, to improve the module? What would be the result?

https://www.oreilly.com/library/view/97-things-every/9780596809515/ch08.html

(Again, It's fine with me as is, I do the same. I skip the migration when it makes the patch harder to understand. But I never skip to add funny comments... :wink: )

Reply:
@elek and here comes the random quote, You can always tag me and I will show up; even if it only for a random quote :facepalm:  

“Without change, something sleeps inside us, and seldom awakens. The sleeper must awaken -- Frank Herbert, Dune "

Thank you for doing your part :), and wake up all.

Reply:
Ok guys I am convinced :) I will move all the Recon SQL DB related configs to Java in this PR. These configs are very good candidates for hiding inside the code.  They are not expected to change often.